core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -519,7 +519,15 @@ def _get_init_args(self):
 
     def hyperparameter_tune(self, scheduler_options, time_limit=None, **kwargs):
         scheduler_options = copy.deepcopy(scheduler_options)
-        scheduler_options[1]['resource'] = self._preprocess_fit_resources(silent=True, **scheduler_options[1]['resource'])
+        resource = copy.deepcopy(scheduler_options[1]['resource'])
+        if 'num_cpus' in resource:
+            if resource['num_cpus'] == 'auto':
+                resource.pop('num_cpus')
+        if 'num_gpus' in resource:
+            if resource['num_gpus'] == 'auto':
+                resource.pop('num_gpus')
```

Comment:
nit: duplicated code between `num_cpus` and `num_gpus` handling, can be put in a small function